front-end/src/api/DashboardApi.js

```diff
@@ -9,3 +9,59 @@ export async function pullUserInfo(userID, token) {
   });
 }
 
+//Attempts to link Github Account
+export async function linkGithubAccount(userID, token, github_username) {
+  const url = process.env.NODE_ENV === 'development' ? 'http://localhost:3000' : 'https://githubconnector.herokuapp.com';
+  const config = {
+    headers:{
+      'Accept': 'application/json',
+      'Content-Type': 'application/json',
+      'Authorization': 'Bearer '  + token }
+  };
+  const payload = {
+    'github_username': github_username
+  };
+  return axios.put(`${url}/users/${userID}/github_info`, payload, config);
+}
+
+
+// Updates Profile Information
```

Comment:
If I'm understanding correctly then yeah I think that would make sense. Like you essentially do an UPDATE [every field] and any old info will just stay the same? I think thats a good idea for consolidating code.

Reply:
@jonabantao, in this case, we have 3 fields that can be already updated with a single PUT request... and those 3 fields could just have 1 save button. I think it would be simpler on all sides to just send those 3 fields in a single request, since the github update path is already separate API call and would still need a separate save button?

Otherwise, I'm not sure the value of checking/parsing fields on the backend that we know can't be updated? (like user id, for example)

Reply:
Unless we are proposing also combining the github username update into the same API call as the standard user update?

Reply:
Things like userId will be fine to be left out since it should be it will be in the path params and validated with our JWT middleware. We can just have the backend query always have the values of email, phone and twitter updated and the frontend has to provide us this information when updating the user profile everytime.

I don't suggest adding the github username to the same PUT request as this updates a different table.

Reply:
Okay. I believe the backend code as it already is should support this...

https://github.com/cs361-spring2020-group8/github-connector/blob/staging/back-end/routes/users.js#L146

If the frontend sends a request body that contains email, twitter, and phone fields, the API will attempt to update all three fields.

Reply:
I get a 500 error when I try to update more than one, so I split it into 3. Maybe I am doing something incorrectly? I will add a video to slack
![editprofile error](https://user-images.githubusercontent.com/7331525/81230723-dd65c980-8fa6-11ea-9f45-ca2abbebcb40.png)